### PR TITLE
Package structure with working proof of concept

### LINKED ISSUE #1: Creating `Rmd` files from code functions
The first issue we raise is the following. Assume that we have an incomplete data set `data` that we would run through a standard test suite to benchmark the scope and state of the missingness and run our pre-imputation diagnostics and tests. 

For example, something close to the following workflow example:
```r
library(mice)
library(dplyr)
library(magrittr)
# missingness over columns
apply(data, 2, function(x) sum(is.na(x)))
# create missing data pattern plot
data %>% 
  md.pattern()
# Some text that discusses the use and interpretation of the pattern
# Some text that discusses the use and interpretation of the pattern
# Some text that discusses the use and interpretation of the pattern
# investigate bivariate data structure
data %>% 
  cor(use = "pairwise.complete.obs")
# investigate missingness relations
# Some text that discusses the use and interpretation of the relations
# Some text that discusses the use and interpretation of the relations
# Some text that discusses the use and interpretation of the relations
data %>% 
  dep.pattern()
# Some text that discusses the use and interpretation of the pattern
# Some text that discusses the use and interpretation of the pattern
# Some text that discusses the use and interpretation of the pattern
# create predictor matrix
data %>% 
  pred
# Some text that discusses the use and interpretation of the predictor matrix
# Some text that discusses the use and interpretation of the predictor matrix
# Some text that discusses the use and interpretation of the predictor matrix
```

**Question 1** How can we realise that the above code is parsed through `dessert(data)` such that the following components are generated:

1. A `Rmd` file that takes as input object `data`
2. A compiled `html` file, with the option to generate additionally a `.pdf` or `.docx` file, cf. `pandoc`
3. [??optionally because of data sharing concerns??] An .Rstudio project with folder that holds the `data` as an `.Rdata` file
4. A `README.md` file that holds information about the `dessert` package, version and function used to generate the archive.

### LINKED ISSUE #2: Find out how to create Rmd files from R code. The idea is to eventually run e.g. dessert(imp) and that an RMD file with all relevant analyses is prepared. Preferably, also compiled.

### LINKED ISSUE #3: When `dessert()` to RMD works, test with simple `t.test()`

### LINKED ISSUE #4: Realize `dessert()` into a separate archive stored locally that contains: a) the .Rmd, b) the .html and .pdf, c) the object as .RData file

### LINKED ISSUE #5: Figure out the working of class-dependent function, such that `dessert(fit.lm)` would run `dessert.lm(fit.lm)`

### PR DESCRIPTION
**Pull request by gerkovink**
Changes:

- `Rmd`'s moved to `inst/rmd` folder to be merged with package install on user machines
- Changed the order of the dessert to first copy to the output directory, and then render --> reason: no write access for most machines on package location
- Changed top level `dessert()` to use `do.call()` and forward an object dependent on its class to `dessert.class()`

### CODE REVIEW

**Comment by gerkovink** [comment]
@hanneoberman, please review.

**Comment by gerkovink** [comment]
@Ernst-Paul please review

File: DESCRIPTION [hunk 1]
```diff
@@ -0,0 +1,24 @@
+Package: dessert
+Type: Package
+Title: Follow your favourite analysis recipe with an automated dessert
```
Comment (hanneoberman) [inline DESCRIPTION]: The title does not yet conform to the R packages advice: [https://r-pkgs.org/Metadata.html#description-title-description](https://r-pkgs.org/Metadata.html#description-title-description)

File: DESCRIPTION [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+License: AGPL (>= 3)
+Encoding: UTF-8
+LazyData: true
+Suggests: 
+    testthat (>= 3.0.0)
+Config/testthat/edition: 3
+RoxygenNote: 7.2.1
+Imports: 
+    rmarkdown,
+    rstudioapi
```
Comment (hanneoberman) [inline DESCRIPTION:24]: I never noticed this import with another package, it it mandatory?

File: R/dessert.R [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+#' @param output_dir The output directory for the rendered output files.
+#'
+#' @return A reproducible standard publication archive
+#' @export
+#'
+#' @examples
+#' fit <- lm(dist ~ speed, data = cars)
+#' dessert(fit)
+dessert <- function(data,
+                    output_format = NULL,
```
Comment (hanneoberman) [inline R/dessert.R:14]: Just checking if this is the right naming convention for the function arguments in the entire package?

File: DESCRIPTION (hunk 2 above)
Comment (hanneoberman) [reply DESCRIPTION:24]: Nevermind! I found out why it's there :)

**Review by hanneoberman — COMMENTED** [review]
Looks good, see in-line comments.

File: DESCRIPTION (hunk 1 above)
Comment (gerkovink) [reply DESCRIPTION]: have updated the title. Let's fill the text more in the coming weeks.

File: R/dessert.R (hunk 3 above)
Comment (gerkovink) [reply R/dessert.R:14]: maybe we'll rename `data` to `input` as a more generic term.

**Pull request merged**